tests/certification/pubsub/azure/eventhubs/eventhubs_test.go

```diff
@@ -0,0 +1,301 @@
+				messageWatcher.ExpectStrings(messages...)
+			}
+
+			// get the sidecar (dapr) client
+			client := sidecar.GetClient(ctx, sidecarName)
+
+			// publish messages
+			ctx.Logf("Publishing messages. sidecarName: %s, topicName: %s", sidecarName, topicName)
+
+			var publishOptions dapr.PublishEventOption
+
+			if metadata != nil {
+				publishOptions = dapr.PublishEventWithMetadata(metadata)
+			}
+
+			for _, message := range messages {
+				ctx.Logf("Publishing: %q", message)
+				var err error
+
+				if publishOptions != nil {
+					err = client.PublishEvent(ctx, pubsubName, topicName, message, publishOptions)
+				} else {
+					err = client.PublishEvent(ctx, pubsubName, topicName, message)
+				}
+
+				require.NoError(ctx, err, "error publishing message")
+			}
+			return nil
+		}
+	}
+
+	assertMessages := func(timeout time.Duration, messageWatchers ...*watcher.Watcher) flow.Runnable {
+		return func(ctx flow.Context) error {
+			// assert for messages
+			for _, m := range messageWatchers {
+				m.Assert(ctx, 25*timeout)
+			}
+
+			return nil
+		}
+	}
+
+	// simulate the publish of messages to iot endpoint (./send-iot-device-events.sh)
+	// addExpectedMessagesforIot := func(messageWatchers *watcher.Watcher) flow.Runnable {
+	// 	return func(ctx flow.Context) error {
+	// 		messages := make([]string, numMessages)
+	// 		for i := range messages {
+	// 			messages[i] = fmt.Sprintf("testmessageForEventHubCertificationTest #%v", i+1)
+	// 		}
+	// 		messageWatchers.ExpectStrings(messages...)
+	// 		return nil
+	// 	}
+	// }
+
+	flow.New(t, "eventhubs certification").
+
+		// Test : single publisher, multiple subscriber with their own consumerID
+		// Run subscriberApplication app1
+		Step(app.Run(appID1, fmt.Sprintf(":%d", appPort),
+			subscriberApplication(appID1, topicName1, consumerGroup1))).
+
+		// Run the Dapr sidecar with the eventhubs component 1.
+		Step(sidecar.Run(sidecarName1,
+			embedded.WithComponentsPath("./components/consumer1"),
+			embedded.WithAppProtocol(runtime.HTTPProtocol, appPort),
+			embedded.WithDaprGRPCPort(runtime.DefaultDaprAPIGRPCPort),
+			embedded.WithDaprHTTPPort(runtime.DefaultDaprHTTPPort),
+			runtime.WithSecretStores(secretStoreComponent),
+			runtime.WithPubSubs(component))).
+
+		// Run subscriberApplication app2
+		Step(app.Run(appID2, fmt.Sprintf(":%d", appPort+portOffset),
+			subscriberApplication(appID2, topicName1, consumerGroup2))).
+
+		// Run the Dapr sidecar with the component 2.
+		Step(sidecar.Run(sidecarName2,
+			embedded.WithComponentsPath("./components/consumer2"),
+			embedded.WithAppProtocol(runtime.HTTPProtocol, appPort+portOffset),
+			embedded.WithDaprGRPCPort(runtime.DefaultDaprAPIGRPCPort+portOffset),
+			embedded.WithDaprHTTPPort(runtime.DefaultDaprHTTPPort+portOffset),
+			embedded.WithProfilePort(runtime.DefaultProfilePort+portOffset),
+			runtime.WithSecretStores(secretStoreComponent),
+			runtime.WithPubSubs(component))).
+
+		// Run dummy app
```

Comment:
A dummy app is not needed ideally to publish to another topic. 
It does not matter to the publisher whether the message is consumed or not ... 
You can make use of the same sidecar sidecar1 itself to publish to another topic which has no subscribers ... Why is this needed?

Reply:
Fixed.